components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -24,25 +25,31 @@ import SaveChangesPanel from '../../save_changes_panel';
 import {ChannelModes} from './channel_modes';
 import {ChannelGroups} from './channel_groups';
 import {ChannelProfile} from './channel_profile';
+import ChannelModeration from './channel_moderation';
 
 interface ChannelDetailsProps {
     channelID: string;
     channel: Channel;
     team: Partial<Team>;
     groups: Group[];
     totalGroups: number;
+    channelPermissions?: Array<ChannelPermissions>;
     allGroups: {[gid: string]: Group}; // hashmap of groups
+    teamScheme?: Scheme;
     actions: {
         getGroups: (channelID: string, q?: string, page?: number, perPage?: number) => Promise<Partial<Group>[]>;
         linkGroupSyncable: (groupID: string, syncableID: string, syncableType: string, patch: Partial<SyncablePatch>) => ActionFunc|ActionResult;
         unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: string) => ActionFunc;
         membersMinusGroupMembers: (channelID: string, groupIDs: Array<string>, page?: number, perPage?: number) => ActionFunc|ActionResult;
         setNavigationBlocked: (blocked: boolean) => any;
         getChannel: (channelId: string) => ActionFunc;
-        getTeam: (teamId: string) => ActionFunc;
+        getTeam: (teamId: string) => any;
```

Comment:
shouldnt this be 
```suggestion
        getTeam: (teamId: string) => ActionFunc;
```

Any reason it was changed?

Reply:
Ah cause you are now calling `.then()` on it  in which case it should be 
```suggestion
        getTeam: (teamId: string) =>  Promise<ActionResult>;
```

Reply:
Thanks! Done 😄